Index: work/crypto/rand/rand_win.c
===================================================================
--- work.orig/crypto/rand/rand_win.c
+++ work/crypto/rand/rand_win.c
@@ -19,7 +19,7 @@
 
 # include <windows.h>
 /* On Windows Vista or higher use BCrypt instead of the legacy CryptoAPI */
-# if defined(_MSC_VER) && _MSC_VER > 1500 /* 1500 = Visual Studio 2008 */ \
+# if defined(_MSC_VER) && _MSC_VER >= 19000 /* 19000 = Visual Studio 2015 */ \
      && defined(_WIN32_WINNT) && _WIN32_WINNT >= 0x0600
 #  define USE_BCRYPTGENRANDOM
 # endif
